### DISCUSSION TITLE
Class token

### DISCUSSION THREAD

**Discussion by vachagancode** [mrdbourke/pytorch-deep-learning#1155] [General]
One  thing that interests me the most is how does the class token we pass in learns from the other patches ?

**Comment by Prezzo-K** [comment]
It can tricky to understand at first go but as the model processes the sequences of image patches  through it attention layers, the class token interacts with other patches and gathers information representing the global context of the entire image.

Eventually it contains a rich, condensed information about what the image is which is then used for classification.
